net/wifidog/patches/010-use-tls-1.2.patch

```diff
@@ -0,0 +1,38 @@
+diff --git a/configure.in b/configure.in
+index bf5463a..43ec27c 100644
+--- a/configure.in
++++ b/configure.in
+@@ -96,8 +96,8 @@ if test "x$enable_cyassl" = xyes; then
+         # the use the new naming scheme below as cyassl/ssl.h is not available for
+         # AC_SEARCH_LIBS
+         AC_CHECK_HEADERS(cyassl/ssl.h)
+-        AC_SEARCH_LIBS([CyaTLSv1_client_method], [cyassl], [], [
+-            AC_SEARCH_LIBS([wolfTLSv1_client_method], [wolfssl], [], [
++        AC_SEARCH_LIBS([CyaTLSv1_2_client_method], [cyassl], [], [
++            AC_SEARCH_LIBS([wolfTLSv1_2_client_method], [wolfssl], [], [
+                             AC_MSG_ERROR([unable to locate SSL lib: either wolfSSL or CyaSSL needed.])
+             ])
+         ])
+@@ -110,7 +110,7 @@ if test "x$enable_cyassl" = xyes; then
+         ]], [[
+                 CYASSL_CTX *ctx;
+                 CyaSSL_Init();
+-                ctx = CyaSSL_CTX_new(CyaTLSv1_client_method());
++                ctx = CyaSSL_CTX_new(CyaTLSv1_2_client_method());
```

Comment:
Two things to note

 - Are TLSv1.0 and TLSv1.1 officially deprecated by IETF?
 - If they are indeed deprecated, we may want to call `wolfSSLv23_client_method()`, `wolfSSL_CTX_SetMinVersion()` so that TLSv1.3 and future versions (expected to be more advanced and secure) can be use.

Please consider presenting the change to upstream first.  Thank you.

Reply:
Draft: https://tools.ietf.org/id/draft-moriarty-tls-oldversions-diediedie-00.html

Too big of a patch. I think upstream is dead. Maybe they moved to wifidog-ng.

Deprecation of TLS 1.2 will take probably a decade.

Reply:
The referred to document is an expired draft to be approved.

I think it's important to allow TLSv1.1 when possible to retain availability as much as possible.  It's too early to break it as a client.

Reply:
> The referred to document is an expired draft to be approved.
> 
> I think it's important to allow TLSv1.1 when possible to retain availability as much as possible. It's too early to break it as a client.

+1

Reply:
Well, choose whichever solution you want. I can't update the pull request as I bricked my laptop (bad BIOS flash).

I still believe TLS 1.2 to be the correct solution. OpenSSL seems to have added support for both in 1.0.1 at the same time. I don't think there are many clients that support 1.1 but not 1.2. SSLLabs seems to confirm this with its client simulations.

The main reason for deprecating TLS 1.1 seems to be the lack of AEAD modes in it.

Reply:
If I understand the ssl/tls right, the protocol negotiation depends on support status of both parties.  If the server offers only TLS1.1, then a client offers only TLS1.2 will fail to connect.  If the server offers TLS1.2 and TLS1.3, we will need to add the setminver call to take advantage of the better TLS1.3.

Reply:
Yep. However upstream mistakenly replaced the SSL_v23 call with TLSv1 in the belief that it's 1.0 and up. Hence the comment in the commit message.

Reply:
Found it: https://github.com/wifidog/wifidog-gateway/commit/fd76bfe555f22fd8ffa5316eb56d46ce24989211